addons/SimpleTODO/TODOItem.gd

```diff
@@ -4,18 +4,184 @@ extends HBoxContainer
+	for i in main.column_container.get_child_count():
+		var child = main.column_container.get_child(i)
+		var rect: Rect2 = child.get_rect()
+		if rect.has_point(mouse_position):
+			return child
+	return null
+
+func get_column_item_from_mouse_position():
+	var column_under_mouse = get_column_from_mouse_position()
+	if column_under_mouse:
+		var column_items = column_under_mouse.item_container
+		var mouse_position = column_items.get_local_mouse_position()
+		
+		for i in column_items.get_child_count():
+			var child: Control = column_items.get_child(i)
+			
+			var rect: Rect2 = child.get_rect()
+			var top_rect: Rect2 = Rect2(rect.position - Vector2(0, item_margin), Vector2(rect.size.x, item_margin))
+			var bottom_rect: Rect2 = Rect2(rect.position + Vector2(0, item_margin), Vector2(rect.size.x, item_margin))
+			
+			if top_rect.has_point(mouse_position) and i <= 0:
+				return {"item": child, "column": column_under_mouse}
+				
+			if bottom_rect.has_point(mouse_position):
+				return {"item": child, "column": column_under_mouse}
+		
+		# Likely we are dragging into a column with no items
+		return {"item": null, "column": column_under_mouse}
+	else:
+		return null
+
+# Handles left click being pressed on the drag panel
+func _on_DragPanel_gui_input(event):
+	if event is InputEventMouseButton:
+		if event.button_index == 1 and event.pressed and not is_dragging:
+			initial_item_index = get_index()
+			
+			get_parent().remove_child(self)
+			
+			parent_column.update_counter()
+			
+			rect_min_size = rect_size
+			
+			# Set the size vertical flags to none so that it doesn't stretch
+			# when being dragged
+			size_flags_vertical = 0
+			
+			main.add_child(self)
+			
+			# Set dragging to true to tell _process to now handle dragging
+			is_dragging = true
+
+			item_placement_holder.get_parent().remove_child(item_placement_holder)
+			parent_column.item_container.add_child(item_placement_holder)
+			item_placement_holder.get_parent().move_child(item_placement_holder, initial_item_index)
+			
+			item_placement_holder.visible = true
+			item_placement_holder.rect_min_size = rect_min_size
+			item_placement_holder.rect_size = rect_size
+
+# Handles left click being released
+func _input(event):
+	if event is InputEventMouseButton:
+		if event.button_index == 1 and not event.pressed and is_dragging:
+			# Set dragging to false to tell _process that we are no
+			# longer dragging this todo item
+			is_dragging = false
+			
+			get_parent().remove_child(self)
+			
+			size_flags_vertical = SIZE_FILL
+
+			rect_min_size = Vector2.ZERO
+			
+			if item_placement_holder:
+				item_placement_holder.visible = false
+				item_placement_holder.get_parent().remove_child(item_placement_holder)
+				main.add_child(item_placement_holder)
+
+			move_item(current_drag_item_index)
+
+			current_drag_item_index = 0
+			initial_item_index = 0
+
+# Used for undo redo, if we can't remove this item then don't error out
```

Comment:
Not sure when this could happen.

Reply:
In my code, there was certain cases with the undo redo system where it would attempt to remove itself when it wasn't attached to any parent which would print errors in the console.